packages/core/src/application.ts

```diff
@@ -147,6 +147,12 @@ export class Application extends Context {
       const options = OPENAPI_SPEC_MAPPING[request.url];
       return this._serveOpenApiSpec(request, response, options);
     }
+
+    if (request.method === 'GET' && request.url === '/explorer') {
+      response.setHeader('Access-Control-Allow-Origin', '*');
```

Comment:
IMO, we need to set Access-Control headers inside `_serveOpenApiSpec`, not here. `GET /explorer` would be invoked from the browser as a regular request for an HTML page.

Reply:
I think the purpose of the story was to allow CORS for all endpoints defined by an application's OpenAPI spec, so that users can test them out using the SwaggerUI instance. With that in mind, I think it should still be set in this method, otherwise it will only be allowed for the `/swagger.{json,yaml}` or `/openapi.{json,yaml}` endpoints. Two things I'm not sure of though are if there are any security implications by doing so, and if we should allow it to be set by our application developers.